feature/component/chip

abdo-essam opened the pull request.

abdo-essam committed:
feat: add NovixChip component and ScrollableChipGroup for enhanced UI

abdo-essam committed:
refactor: simplify NovixChip component by removing variant handling

marwanqashwa commented on app/src/main/java/com/london/novix/presentation/component/NovixChip.kt:
```diff
@@ -0,0 +1,98 @@
+    onClick: () -> Unit = {}
+) {
+    val colors = NovixTheme.colors
+    val typography = NovixTheme.typography
```
i think use color or typography direct not get all will be better

marwanqashwa commented on app/src/main/java/com/london/novix/presentation/component/NovixChip.kt:
```diff
@@ -0,0 +1,98 @@
+    val typography = NovixTheme.typography
+
+    val backgroundColor = if (isSelected) colors.secondary else colors.surface
+    val textColor = if (isSelected) colors.onPrimary else colors.body
```
i think you can use animate color as state will be better than direct if

marwanqashwa commented on app/src/main/java/com/london/novix/presentation/component/NovixChip.kt:
```diff
@@ -0,0 +1,98 @@
+    val textColor = if (isSelected) colors.onPrimary else colors.body
+    val shape = if (isSelected) RoundedCornerShape(12.dp) else RoundedCornerShape(8.dp)
+    val verticalPadding = 8.dp
+    val horizontalPadding = if (isSelected) 24.dp else 12.dp
```
same here you can animate size

marwanqashwa commented on app/src/main/java/com/london/novix/presentation/component/NovixChip.kt:
```diff
@@ -0,0 +1,98 @@
+}
+
+@Composable
+fun ScrollableChipGroup(
```
make it private if it for review and add subfix preview

marwanqashwa reviewed (commented):
great job just take a look at a few comment i left

abdo-essam committed:
feat: add animations to NovixChip component for improved interactivity

abdo-essam committed:
Merge branch 'develop' of https://github.com/LondonSquad/Novix into feature/component/chip

Abdulrahman-Ragab-01 commented on app/src/main/java/com/london/novix/presentation/component/NovixChip.kt:
```diff
@@ -0,0 +1,120 @@
+        label = "horizontal_padding"
+    )
+
+    Row(
```
I think there is no need to use Row, you can just put the modifier in the Text component itself

ahmeddsamii replied:
i agree with abdulrahman

ahmeddsamii commented on app/src/main/java/com/london/novix/presentation/component/NovixChip.kt:
```diff
@@ -0,0 +1,120 @@
+        modifier = modifier,
+        horizontalArrangement = Arrangement.spacedBy(8.dp)
+    ) {
+        items(chips.size) { index ->
```
you can use the item itself by using items(chips) unless if you need the index and in this case i don't see that you are using the index

abdo-essam committed:
feat(ui): Improve NovixChip composable

Refactor NovixChip for better readability and flexibility.
- Removed unnecessary Row composable and simplified the layout.
- Directly used Text composable with required modifiers.
- Used `items` lambda to iterate over chips list directly in
  `ChipsRow` composable.

ahmeddsamii reviewed (approved):
Nice job bro

Abdulrahman-Ragab-01 reviewed (approved):
Great Job!
